driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -35,15 +40,40 @@ struct Context {
 impl UpdatingOrderbook {
     /// Does not block on initializing the orderbook. This will happen in the first call to
     /// `get_auction_data` which can thus take a long time to complete.
-    pub fn new(contract: Arc<dyn StableXContract + Send + Sync>, web3: Web3) -> Self {
+    pub fn new(
+        contract: Arc<dyn StableXContract + Send + Sync>,
+        web3: Web3,
+        path: Option<PathBuf>,
+    ) -> Self {
+        // Recover the orderbook from file (if possible, otherwise use default).
+        let orderbook = match &path {
+            Some(path) => match Orderbook::try_from(path.as_path()) {
+                Ok(orderbook) => {
+                    info!("successfully recovered orderbook from path");
+                    orderbook
+                }
+                Err(error) => {
+                    // This will always happen when file does not exist (i.e. on first startup)
+                    // TODO: match on this error and don't warn when file doesn't exist.
+                    warn!(
+                        "Failed to construct orderbook from path (using default): {}",
+                        error
+                    );
```

Comment:
```suggestion
                    if path.exists() {
                        warn!(
                            "Failed to construct orderbook from path (using default): {}",
                            error
                        );
                    }
```

Reply:
SOrry @nlordell ... too late. Will make a new PR with this change.

Reply:
I suppose this is cheeky way to determine that the failure was not because the file did not exist?

Reply:
It checks if the file exists before logging the warning.

Reply:
Addressed in #817